qmail-send.c

```diff
@@ -173,14 +173,14 @@ char *recip;
    if (str_equal(sender + i - 4,"-@[]"))
     {
      unsigned int j = byte_rchr(sender,i - 4,'@');
-     unsigned int k = str_rchr(recip,'@');
-     if (recip[k] && (j + 5 <= i))
+     char *at = strrchr(recip,'@');
+     if (at && (j + 5 <= i))
```

Comment:
I had a moment of doubt, but indeed: `str_rchr(recip,'@') == '\0'` is the same as `strrchr(recip,'@') == NULL`.